work/load/load_processors.go

```diff
@@ -0,0 +1,175 @@
+	return quantity
+}
+
+func (p *loadProcessor) Frequency() time.Duration {
+	return frequency
+}
+
+func (p *loadProcessor) getProcessResult(wrk *work.Work, result any, errMsg *string) *work.ProcessResult {
+	switch result.(string) {
+	case work.ResultFailing:
+
+		msg := "failing from load test"
+		if errMsg != nil {
+			msg = *errMsg
+		}
+
+		return work.NewProcessResultFailing(work.FailingUpdate{
+			FailingError: *errors.NewSerializable(errors.New(msg)),
+			Metadata:     wrk.Metadata,
+		})
+	case work.ResultFailed:
+		return work.NewProcessResultFailed(work.FailedUpdate{
+			FailedError: *errors.NewSerializable(errors.New("failed from load test")),
+			Metadata:    wrk.Metadata,
+		})
+	case work.ResultSuccess:
+		return work.NewProcessResultSuccess(work.SuccessUpdate{
+			Metadata: wrk.Metadata,
+		})
+	default:
+		return work.NewProcessResultDelete()
+	}
+}
+
+func (p *loadProcessor) chooseDopeyResult(wrk *work.Work) string {
+	possibleWorkResults := []string{
+		work.ResultDelete,
+		work.ResultFailing,
+		work.ResultFailed,
+		work.ResultSuccess,
+	}
+	if result := wrk.Metadata[MetadataProcessResult]; result != nil {
+		if resultStr, ok := result.(string); ok {
+			if resultStr == "random" {
+				var indexnum int = rand.IntN(len(possibleWorkResults) - 1)
+				return possibleWorkResults[indexnum]
+			}
+			if slices.Contains(possibleWorkResults, resultStr) {
+				return resultStr
+			}
+		}
+	}
+	return work.ResultDelete
+}
+
+func (p *loadProcessor) Process(ctx context.Context, wrk *work.Work, updater work.ProcessingUpdater) work.ProcessResult {
+	switch wrk.Type {
+	case TypeSleepy:
+		waitMillis := rand.IntN(500)
+		time.Sleep(time.Duration(waitMillis) * time.Millisecond)
+		return *p.getProcessResult(wrk, work.ResultSuccess, nil)
+	case TypeDopey:
+		return *p.getProcessResult(wrk, p.chooseDopeyResult(wrk), nil)
+	default:
+		return *p.getProcessResult(wrk, work.ResultFailed, pointer.FromString(fmt.Sprintf("[%s] not an expected work type", wrk.Type)))
+	}
+}
+
+type LoadItem struct {
+	SecondsOffsetFromStart int64        `json:"secondsOffsetFromStart"`
+	Create                 *work.Create `json:"create"`
+}
+
+func NewLoadWorkCreate(create *work.Create) (*work.Create, error) {
+	if create == nil {
+		return nil, errors.New("create is missing")
+	}
+	if create.Type != TypeSleepy && create.Type != TypeDopey {
+		return nil, fmt.Errorf("invalid work type %s", create.Type)
+	}
+
+	availableTime := time.Now().Add(processingAvailableTimeDelay)
+	if !create.ProcessingAvailableTime.IsZero() {
+		availableTime = create.ProcessingAvailableTime.Add(processingAvailableTimeDelay)
+	}
+	timeout := processingTimeout
+	if create.ProcessingTimeout != 0 {
+		timeout = create.ProcessingTimeout
+	}
+
+	metadata := map[string]any{}
+
+	if create.Metadata != nil && create.Metadata[MetadataProcessResult] != nil {
+		metadata[MetadataProcessResult] = create.Metadata[MetadataProcessResult]
+	}
+
+	return &work.Create{
```

Comment:
Why aren't you just straight-away using the incoming `create` directly? The caller should be setting the `ProcessingAvailableTime` based upon the `SecondsOffsetFromStart`. I suppose if the ProcessingTime is 0, then you could just change it to the default, which rather that 5 minutes, I'd make infinite. We want to see if we can get it stuck.

Reply:
In fact, why do we even need this function at all? Couldn't we send the `create` from the parsed JSON directly into the work system?